src/ui-components/input-field/input-field.js

```diff
@@ -0,0 +1,106 @@
+import React, {Component} from "react";
+import "./input-field.scss";
+import { CSS_CLASSES } from "./constants";
+
+class InputField extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isInputFieldFocused: false,
+      value: this._getInitialValue(props.value, props.type)
+    };
+  }
+
+  _getInitialValue(providedValue, inputType) {
+    let value = providedValue;
+
+    if(typeof providedValue === "undefined") {
+      value = "";
+    }    
+    if(inputType !== "number" && typeof providedValue === "number") {
+      value = providedValue.toString();
+    } 
+    if(inputType === "number" && typeof providedValue !== "number") {
+      value = "";
+    }    
+    return value;
+  }
+
+  _getFieldType(providedType) {
+    let type = "text";
+
+    if(providedType === "number") {
+      type = providedType;
+    }
+    if(providedType === "password") {
+      type = providedType
+    }
+
+    return type;
+  }
+
+  _handleChange(event) {
+    this.setState({ value: event.target.value })
+    event.stopPropagation();
+  }
+
+  _handleFocus(event) {
+    this.setState({ isInputFieldFocused: true });
+    event.stopPropagation();
+  }
+
+  _handleBlur(event) {
+    this.setState({ isInputFieldFocused: false });
+    event.stopPropagation();
+  }
+
+  _getLabelClassName() {
+    let className = `${CSS_CLASSES.LABEL}`;
+    if (this.state.value || this.state.isInputFieldFocused) {
+      className = `${className} ${CSS_CLASSES.LABEL_RAISED}`
+
+      if (this.state.isInputFieldFocused) {
+        className = `${className} ${CSS_CLASSES.LABEL_FOCUSED}`;
+      }
+    } else {
+      className = `${className} ${CSS_CLASSES.LABEL_LOWERED}`;
+    }
+
+    return className;
+  }
+
+  render() {
+    const {
+      id,
+      type,
+      required,
+      label,
+      validationMessages = [],
+    } = this.props;
+
+    return (
+      <div className={CSS_CLASSES.INPUT_WRAPPER}>
+        {label && <label className={this._getLabelClassName()} htmlFor={id}> {label} </label>}
+        <input
+          id={id}
+          className={CSS_CLASSES.FIELD}
+          type={this._getFieldType(type)}
+          required={required}
+          value={this.state.value}
+          onChange={(event) => this._handleChange(event)}
+          onFocus={(event) => this._handleFocus(event)}
+          onBlur={(event) => this._handleBlur(event)}
+        />
+        {validationMessages.length > 0 &&
```

Comment:
Preference thing, but just checking for `validationMessages.length` is enough :D

Reply:
Not really - in this case, react does a check-up and prints out the value which happens to be 0. Refuted.